File: 1. Two Sum/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        p = 0
        for i in range(len(nums)):
            for j in range(1, len(nums)):
                if ((nums[i] + nums[j] == target) and i != j):
                    ans = [i, j]
                    p = 1
                    break
            if p == 1:
                break
        return ans

File: 1. Two Sum/1. Two Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let clues = {};

    for (let i = 0; i < nums.length; i++) {        
        
        // if a previous element needs this value
        if (nums[i] in clues) {
            return [clues[nums[i]], i];
        }

        // save the clue to check later
        clues[target - nums[i]] = i;
    }
    return [];
};